status_and_problems.tex

```diff
@@ -0,0 +1,134 @@
+\end{itemize}
+
+
+\subsubsection{Distribution System} \label{sec:statusDistrib}
+
+The main tool used for distribution is {\bf Docker}. 
+
+In \url{pipelines.lsst.io} instructions are provided on how to deploy a science pipelines docker image.
+
+A second distribution approach is also provided, using the script \textit{newinstall.sh}. 
+This scripts permits to retrieve a specific version (or reference) of the science pipelines from the Eups repository, and setup the environment for its build.
+
+
+\subsubsection{SW Products Identification} \label{sec:statusIdentification}
+
+The software products are developed using multiple GitHub packages. 
+A GitHub metapackage, for example \textit{lsst\_distrib} for science pipelines, is identifying the software product. 
+All git packages of a software product are dependencies in that metapackage, direct or indirect.
+
+GitHub teams are used to differentiate between two types of git packages:
+
+\begin{itemize}
+\item {\bf Data Management}: it includes all dependencies of \textit{lsst\_distrib}
+\item {\bf DM Externals}: for packages used to build and distribute \textit{lsst\_distrib} but that they are not part of it.
+\end{itemize}
+
+There is a third team, {\bf DM Auxiliary}, that identify auxiliary packages to be tagged, but that they are not part of the software product:
+
+Note that in reality, \textit{lsst\_distrib} do not correspond to a software product, but to a distribution product.
+It includes not only the different pipelines, but also test data and build tools.
+
+
+\subsubsection{Environment} \label{sec:statusEnvs}
+
+The conda environment to be used for building the science pipelines is defined in the Git package \textit{scipipe\_conda\_env}.
+No releases nor tags are made  on the conda environment git package.
+
+\textit{TO BE CLARIFIED: how are java build environments managed?}
+
+
+\subsubsection{Build Tools} \label{sec:statusTools}
+
+The build tools used to build science pipelines software are available in \textit{lsstsw} and \textit{lsst\_build} git packages.
+
+The Jenkins scripts also are an important part of the tooling, facilitating a large quantity of actions, like, test a ticket branch, automate releases, or provide periodic build (weekly/daily).
+
+The tools picture is completed by \textit{codekit}, that permits to interact with multiple GitHub packages at the same time.
+Just as an example, it is used to create a tag on all the GitHub packages composing a defined product.
+
+\textit{TO BE CLARIFIED: which are java build tools?}
+
+
+\subsection{Open Problems} \label{sec:openProblems}
+
+\subsubsection{SW Product Identification} \label{sec:problemId}
+
+The definition of software product is not unique.
+
+A GitHub package may be included in more than one SW product.
+This make impossible to apply in a consistent way the release procedure to other SW products. It can be applied only to release the science pipelines.
```

Comment:
This makes it impossible to apply a release procedure in a consistent manner to multiple software products.  

I think you mean the current release procedure - we are trying to have one we can apply consistently

Reply:
The release procedure we have can be applied only to one software product: science pipelines. We can't apply it to dax_imgserver for example. The only thing we can do is to release all DM software at once.